pkg/decoders/escaped_unicode.go

```diff
@@ -112,33 +192,87 @@ func decodeCodePoint(input []byte) []byte {
 }
 
 func decodeEscaped(input []byte) []byte {
-	// Find all Unicode escape sequences in the input byte slice
-	indices := escapePat.FindAllSubmatchIndex(input, -1)
+	return decodeWithPattern(input, escapePat)
+}
 
-	// Iterate over found indices in reverse order to avoid modifying the slice length
-	utf8Bytes := make([]byte, maxBytesPerRune)
-	for i := len(indices) - 1; i >= 0; i-- {
-		matches := indices[i]
-		startIndex := matches[0]
-		hexStartIndex := matches[2]
-		endIndex := matches[3]
+// decodeBraceEscape handles \u{X} format - Rust, Swift, some JS, etc.
+func decodeBraceEscape(input []byte) []byte {
+	return decodeWithPattern(input, braceEscapePat)
+}
 
-		// Extract the hexadecimal value from the escape sequence
-		hexValue := string(input[hexStartIndex:endIndex])
+// decodeLongEscape handles \U00XXXXXX format - Python, etc.
+func decodeLongEscape(input []byte) []byte {
+	return decodeWithPattern(input, longEscapePat)
+}
 
-		// Parse the hexadecimal value to an integer
-		unicodeInt, err := strconv.ParseInt(hexValue, 16, 32)
-		if err != nil {
-			// If there's an error, continue to the next escape sequence
-			continue
-		}
+// decodePerlEscape handles \x{X} format - Perl
+func decodePerlEscape(input []byte) []byte {
+	return decodeWithPattern(input, perlEscapePat)
+}
 
-		// Convert the Unicode code point to a UTF-8 representation
-		utf8Len := utf8.EncodeRune(utf8Bytes, rune(unicodeInt))
+// decodeCssEscape handles \X format - CSS (hex without padding, with space delimiter or end of string or next hex sequence)
+func decodeCssEscape(input []byte) []byte {
+	return decodeWithPattern(input, cssEscapePat)
+}
 
-		// Replace the escape sequence with the UTF-8 representation
-		input = append(input[:startIndex], append(utf8Bytes[:utf8Len], input[endIndex:]...)...)
-	}
+// decodeHtmlEscape handles &#xX; format - HTML/XML
+func decodeHtmlEscape(input []byte) []byte {
+	return decodeWithPattern(input, htmlEscapePat)
+}
 
-	return input
+// decodePercentEscape handles %uXXXX format - Percent-encoding (non-standard)
+func decodePercentEscape(input []byte) []byte {
+	return decodeWithPattern(input, percentEscapePat)
 }
+
+// decodeHexEscape handles 0xX format - Hexadecimal notation with space separation
+// func decodeHexEscape(input []byte) []byte {
```

Comment:
Are we intentionally commenting out this code?

Reply:
Ah, I don’t want to be a blocker, but personally I’m not a big fan of leaving large chunks of code commented out 😅